api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -193,6 +193,18 @@ func (mw *azureManagedControlPlaneWebhook) ValidateUpdate(ctx context.Context, o
 		allErrs = append(allErrs, err)
 	}
 
+	if !reflect.DeepEqual(m.Spec.AzureEnvironment, old.Spec.AzureEnvironment) {
+		old.setAzureEnvironmentDefault()
```

Comment:
since validation webhooks run after defaulting webhooks, you shouldn't need to do this here

Instead, you can use the same pattern that's used everywhere else in this file to validate immutability with `webhookutils.ValidateImmutable()` (see lines above for examples)